### DISCUSSION TITLE
higher precision reduce

### DISCUSSION THREAD

**Discussion by jcupitt** [libvips/libvips#2047] [Ideas]
The vector path currently uses a 16 x 16 -> 32 mul followed by a shift. We should experiment with 32 x 32 -> to 32 or 64. We could use the same precision as the C path.

See:

https://github.com/libvips/pyvips/issues/148#issuecomment-570721063

Similarly for the conv vector path.

**Comment by homm** [comment]
> The vector path currently uses a 16 x 16 -> 32 mul followed by a shift

It's not clear for me how this fits in the previous statement «2.6 bits for the coefficients, 10.6 for the accumulator». 16 x 16 -> 32 is totally enough for 8-bit sources, while «2.6 bits for the coefficients» is not.

**Comment by kleisauke** [comment]
This should be fixed after PR #3618, see:
https://www.libvips.org/2023/10/10/What's-new-in-8.15.html#performance-improvements
